libraries/chain/include/eosio/chain/webassembly/eos-vm-oc.hpp

```diff
@@ -0,0 +1,818 @@
+#pragma once
+
+#include <eosio/chain/webassembly/common.hpp>
+#include <eosio/chain/exceptions.hpp>
+#include <eosio/chain/webassembly/runtime_interface.hpp>
+#include <eosio/chain/apply_context.hpp>
+#include <softfloat.hpp>
+#include "IR/Types.h"
+
+#include <eosio/chain/webassembly/eos-vm-oc/eos-vm-oc.hpp>
+#include <eosio/chain/webassembly/eos-vm-oc/memory.hpp>
+#include <eosio/chain/webassembly/eos-vm-oc/executor.hpp>
+#include <eosio/chain/webassembly/eos-vm-oc/code_cache.hpp>
+#include <eosio/chain/webassembly/eos-vm-oc/config.hpp>
+#include <eosio/chain/webassembly/eos-vm-oc/intrinsic.hpp>
+
+#include <boost/hana/equal.hpp>
+
+namespace eosio { namespace chain { namespace webassembly { namespace eosvmoc {
+
+using namespace IR;
+using namespace Runtime;
+using namespace fc;
+using namespace eosio::chain::webassembly::common;
+
+using namespace eosio::chain::eosvmoc;
+
+class eosvmoc_instantiated_module;
+
+class eosvmoc_runtime : public eosio::chain::wasm_runtime_interface {
+   public:
+      eosvmoc_runtime(const boost::filesystem::path data_dir, const eosvmoc::config& eosvmoc_config, const chainbase::database& db);
+      ~eosvmoc_runtime();
+      bool inject_module(IR::Module&) override { return false; }
```

Comment:
Doesn't this need injection?

Reply:
Injection is done in run_compile(), this avoids the post-injection/re-injection function size differences